### DISCUSSION TITLE
Best Practices For Insta Filtering/Redactions

### DISCUSSION THREAD

**Discussion by NicholasLYang** [mitsuhiko/insta#674] [Q&A]
Hi, we're trying out Insta in [turborepo](http://github.com/vercel/turborepo). We currently use Prysk, which lets you write inline regexes into your snapshots ([example](https://github.com/vercel/turborepo/blob/1b1f98edd0945c3748e54e89c2a92b7415530bac/turborepo-tests/integration/tests/turbo-trace.t#L33)). This is pretty convenient for stuff like normalizing paths across platforms.

I'm wondering what the equivalent solution would look like in Insta. I tried doing a filter that just replaces `\` with `/`, but that isn't great because not all occurrences of `\` are in paths. I can try to manually normalize the data before it reaches Insta, but that's a little tedious tbh. I could also split snapshots across OS, so I get a `turbo_test_windows.snap` and a `turbo_test_mac.snap`, but that makes updating snapshots difficult, especially across operating systems.

To be clear, this isn't a feature request. I'm just curious what the idiomatic "Insta-y" solution would be.

Thanks!

**Comment by max-sixty** [comment]
Possibly dynamic redactions? https://insta.rs/docs/redactions/ Then we can use a regex / any custom logic...

**Reply by NicholasLYang** [reply]
Hmm okay. That makes sense for structured data, but perhaps not for larger blocks of text. I'll give it a shot though!
